include/verilated_vpi.cpp

```diff
@@ -681,14 +687,80 @@ class VerilatedVpiPutHolder final {
             }
             default: break;
             }
-            m_vector.resize(words);
-            std::memcpy(m_vector.data(), valuep->value.vector, words * sizeof(s_vpi_vecval));
-            m_value.value.vector = m_vector.data();
+            new (&m_storage.vec)
+                std::vector<s_vpi_vecval>{valuep->value.vector, &valuep->value.vector[words]};
+            m_value.value.vector = m_storage.vec.data();
             break;
         }
         }
     }
 
+    VerilatedVpiPutHolder(VerilatedVpiPutHolder const& o)
+        : m_var{o.m_var}
+        , m_value{o.m_value} {
+        switch (m_value.format) {
+        case vpiBinStrVal:
+        case vpiOctStrVal:
+        case vpiDecStrVal:
+        case vpiHexStrVal:
```

Comment:
Similar below please.  What you have currently is FALLTHRU to vpiStringVal handling, which seems wrong I presume.

```suggestion
        case vpiBinStrVal: ;
        case vpiOctStrVal: ;
        case vpiDecStrVal: ;
        case vpiHexStrVal: ;
```

Reply:
That's still fallthru? But we do want fallthru, all those types use the `str` field.

Reply:
Please add 

   // FALLTHRU

to each case line then.